src/oss/langgraph/workflows-agents.mdx

```diff
@@ -0,0 +1,2221 @@
+
+# Invoke
+state = orchestrator_worker.invoke({"topic": "Create a report on LLM scaling laws"})
+
+from IPython.display import Markdown
+Markdown(state["final_report"])
+```
+</Tab>
+<Tab title="Functional API">
+```python
+from typing import List
+
+
+# Schema for structured output to use in planning
+class Section(BaseModel):
+    name: str = Field(
+        description="Name for this section of the report.",
+    )
+    description: str = Field(
+        description="Brief overview of the main topics and concepts to be covered in this section.",
+    )
+
+
+class Sections(BaseModel):
+    sections: List[Section] = Field(
+        description="Sections of the report.",
+    )
+
+
+# Augment the LLM with schema for structured output
+planner = llm.with_structured_output(Sections)
+
+
+@task
+def orchestrator(topic: str):
+    """Orchestrator that generates a plan for the report"""
+    # Generate queries
+    report_sections = planner.invoke(
+        [
+            SystemMessage(content="Generate a plan for the report."),
+            HumanMessage(content=f"Here is the report topic: {topic}"),
+        ]
+    )
+
+    return report_sections.sections
+
+
+@task
+def llm_call(section: Section):
+    """Worker writes a section of the report"""
+
+    # Generate section
+    result = llm.invoke(
+        [
+            SystemMessage(content="Write a report section."),
+            HumanMessage(
+                content=f"Here is the section name: {section.name} and description: {section.description}"
+            ),
+        ]
+    )
+
+    # Write the updated section to completed sections
+    return result.content
+
+
+@task
+def synthesizer(completed_sections: list[str]):
+    """Synthesize full report from sections"""
+    final_report = "\n\n---\n\n".join(completed_sections)
+    return final_report
+
+
+@entrypoint()
+def orchestrator_worker(topic: str):
+    sections = orchestrator(topic).result()
+    section_futures = [llm_call(section) for section in sections]
+    final_report = synthesizer(
+        [section_fut.result() for section_fut in section_futures]
+    ).result()
+    return final_report
+
+# Invoke
+report = orchestrator_worker.invoke("Create a report on LLM scaling laws")
+from IPython.display import Markdown
+Markdown(report)
+```
+</Tab>
+</Tabs>
+:::
+
+:::js
+<Tabs>
+<Tab title="Graph API">
+```typescript
+import { z } from "zod";
+
+// Schema for structured output to use in planning
+const sectionSchema = z.object({
+  name: z.string().describe("Name for this section of the report."),
+  description: z.string().describe(
+    "Brief overview of the main topics and concepts to be covered in this section."
+  ),
+});
+
+const sectionsSchema = z.object({
+  sections: z.array(sectionSchema).describe("Sections of the report."),
+});
+
+// Augment the LLM with schema for structured output
+const planner = llm.withStructuredOutput(sectionsSchema);
+```
+
+### Creating workers in LangGraph
```

Comment:
Oh I see -- it looks like this is duplicated rather than having separate fences. We might want to clean this up